client/fingerprint/consul.go

```diff
@@ -298,6 +303,60 @@ func (cfs *consulFingerprintState) grpcTLSPort(info agentconsul.Self) (string, b
 	return fmt.Sprintf("%d", int(p)), ok
 }
 
+func (cfs *consulFingerprintState) dnsPort(info agentconsul.Self) (string, bool) {
+	p, ok := info["DebugConfig"]["DNSPort"].(float64)
+	return fmt.Sprintf("%d", int(p)), ok
+}
+
+// dnsAddr fingerprints the Consul DNS address, but only if Nomad can use it
+// usefully to provide an iptables rule to a task
+func (cfs *consulFingerprintState) dnsAddr(logger hclog.Logger) func(info agentconsul.Self) (string, bool) {
+	return func(info agentconsul.Self) (string, bool) {
+
+		// only addresses we can use for an iptables rule from a container to the
+		// host will be fingerprinted
+		isValidForTaskUse := func(addr netip.Addr) (string, bool) {
+			if !addr.IsLoopback() && !addr.IsUnspecified() && addr.IsValid() {
+				return addr.String(), true
+			}
+			return "", false
```

Comment:
this is really "6 of one, half a dozen of the other" but my instinct is to reverse the logic here

```suggestion
			if !addr.IsValid() || addr.IsLoopback() || addr.IsUnspecified() {
				return "", false
			}
			return addr.String(), true
```

up to your preference :shrug: